packages/analytics-js/src/app/RudderAnalytics.ts

```diff
@@ -50,55 +51,62 @@
  * consume SDK preload event buffer
  */
 class RudderAnalytics implements IRudderAnalytics<IAnalytics> {
+  // START-NO-SONAR-SCAN
+  // eslint-disable-next-line sonarjs/public-static-readonly
   static globalSingleton: Nullable<RudderAnalytics> = null;
+  // END-NO-SONAR-SCAN
   analyticsInstances: Record<string, IAnalytics> = {};
   defaultAnalyticsKey = '';
   logger = defaultLogger;
 
   // Singleton with constructor bind methods
   constructor() {
-    if (RudderAnalytics.globalSingleton) {
-      // START-NO-SONAR-SCAN
-      // eslint-disable-next-line no-constructor-return
-      return RudderAnalytics.globalSingleton;
-      // END-NO-SONAR-SCAN
+    try {
+      if (RudderAnalytics.globalSingleton) {
+        // START-NO-SONAR-SCAN
+        // eslint-disable-next-line no-constructor-return
+        return RudderAnalytics.globalSingleton;
```

Comment:
This statement is partially true, but there’s an important clarification to make about when returning from a constructor in JavaScript and TypeScript actually affects behavior:

If a constructor returns a non-object (primitive) value (like a number, string, null, or undefined), JavaScript ignores that return value and still returns the instance of the class.

If a constructor returns an object, JavaScript does not ignore the return value. Instead, it returns that object instead of the instance of the class, which can indeed lead to unexpected behavior. In this case, the new operator will return the object that was explicitly returned from the constructor, rather than the instance of the class.